universe/systems.py

```diff
@@ -0,0 +1,78 @@
+from decimal import Decimal
+import random
+
+
+class UpdateSystem:
+    def process(self, manager):
+        for _id, entity in manager.get_components('queue').items():
+            queue = entity._data['queue']
```

Comment:
Thoughts: lots of `entity._data[field_name]` going on in this module that can be simplified to `entity.field_name` after doing the `__getattr__` suggestion in the `Entity` class.